sway-lsp/src/capabilities/formatting.rs

```diff
@@ -18,7 +18,7 @@ pub fn format_document(
 pub fn get_format_text_edits(text: Arc<str>, options: FormattingOptions) -> Option<Vec<TextEdit>> {
     // we only format if code is correct
 
-    match get_formatted_data(text.clone(), options) {
+    match get_formatted_data(text.clone(), options, None) {
```

Comment:
Should this also be passed a `BuildConfig` for correctness? Otherwise it appears `forc fmt` might behave differently from `sway-lsp`'s formatting?

Reply:
That would mean making the lsp aware of a) what file the text is part of b) the manifest (forc.toml) likely associated with said file. With `forc fmt` this isn't that big an issue since it seems intended to run in a filesystem-aware way anyway, with the LSP it might not be the intended behavior (?)

If you think the behavior is fine I'd implement it in an optional way (i.e. if possible to find associated file & manifest pass them to `get_format_text_edits`, if not pass `None`, so you can still format any arbitrary bit of compileable sway code)

Seems like a higher-level design question, so up to you.

Reply:
I've opened up #1283 to track this - happy to see this land in the meantime :)

Reply:
I assume that's #1299

Reply:
Ah woops, yes